### ISSUE TITLE
Downstream sensors button

### ISSUE THREAD

**Issue by nivdror** [apache/airflow#39939]
### Description

Hi,
We have custom sensors that are based on BaseSensorOperator to link between our dags in our Airlow environment.
Right now we have the ability to move backwards via external DAG button when pressing on the sensor in the graph UI.

This is not enough, I would to be able to travel forward as well. I would like the graph to show on which tasks a sensor is listening and tab of connection(with links of course) to those sensors.

We created this feature in our own old airflow but I don't see why this cannot be in apache airflow.
Here is an example:
![Screenshot 2024-05-30 at 12 15 03](https://github.com/apache/airflow/assets/131352704/981d8460-b6b0-4238-8813-b4cc31aeabdc)

after pressing the downstream sensors button
![Screenshot 2024-05-30 at 12 15 46](https://github.com/apache/airflow/assets/131352704/61e44585-95a1-41e0-baf2-61a220165f53)

### Use case/motivation

I'm trying to achieve a better way to move/investigate dags in airflow

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by Taragolis** [comment]
If you would like propose a feature better to provide a PR, otherwise it might take unpredictable time to implement it. In short  until someone who want to implement this and might take a time from couple day to decades.

Apache Airflow is an OSS and everyone could propose changes by raising Pull Request, and if it pass code review it would be include into the next minor version of Airflow.

**Comment by nivdror** [comment]
I'm aware that Airflow is a open source project. However I don't possess any frontend skills nor the time to learn them.
So I open a feature request in hope that someone will think the idea is worthy and implement it.

**Comment by nathadfield** [comment]
@nivdror Maybe you could collaborate with the people who did create this as a custom feature to bring it to the project?

**Comment by nivdror** [comment]
@nathadfield No, those people have left the company

**Comment by nathadfield** [comment]
Ok, well, we'll leave this as an open feature request but it might not get any traction without someone else seeing the value and making it happen.  Maybe you could try to build some connections with other UI interested community members and collaborating from there?

**Comment by nivdror** [comment]
@nathadfield thank you. I will see what I can do

**Comment by eladkal** [comment]
Isn't that exactly what you can see in the dataset view? You can move upstream or downstream

**Comment by nivdror** [comment]
Hi, we won't use datasets.
Our architecture is built mainly on three layers - scheduler layer, context layer and compute layer.
Each got it's dependencies and definition. So we see airflow as scheduler which doesn't need to what it's running(spark, MapReduce etc') nor where (CDH, EMR, Databricks etc') nor the data artifacts we output.

This helps us migrate through different providers of Airflow + plus helps us decouple the dependencies. 
In summary, we don't put all of our eggs in one basket.

**Comment by bbovenzi** [comment]
We did a bit to show datasets connected upstream and downstream of a DAG on the DAG page. It's not a bad idea to include all types of dependencies like sensors and triggers.

**Comment by bbovenzi** [comment]
Duplicate of https://github.com/apache/airflow/issues/49576

**Comment by nivdror** [comment]
@bbovenzi Hi, 
how is this a duplicate of the https://github.com/apache/airflow/issues/49576 PR?

**Comment by bbovenzi** [comment]
> [@bbovenzi](https://github.com/bbovenzi) Hi, how is this a duplicate of the [#49576](https://github.com/apache/airflow/issues/49576) PR?

With the launch of 3.0, we won't be adding any new features to 2.10. But that issue is to better incorporate sensors and triggers in the dag graphs. If there is something missing then please chime in.
I figure that task.doc_md and extra links with a better graph should get you most of the way there?

**Comment by nivdror** [comment]
obviously, I'm not using airflow 3.0 yet, but I understand not wanting to advance 2.10 further.
The main idea of this feature is to allow navigation between adjacent dags.
I don't think the new feature will get there but once we moved airflow 3.* maybe I will create another PR
